/new/proposal form loads if chain data not loaded yet

mhagel opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: #3988 

## Description of Changes
- Inits chain if API not loaded yet on new proposal page
- reusable useInitChain hook

## Test Plan
- ran proposals.spec.ts e2e tests
- CA (click around) tested on local:
  - http://localhost:8080/evmos/new/proposal shows "Feature not supported yet for this community"
      - loads on direct url 
      - loads from "Create onchain proposal" click 
  - repeat for any other chain: loads Proposal form
  - Click around between multiple communities in sidebar
     - sometimes click to Proposals first
     - sometimes click to New Onchain Proposal first

## Deployment Plan
<!--- Omit if unneeded -->
1. 

## Other Considerations
<!--- Follow-up tickets, breaking changes, etc -->
- Evmos still won't be able to create proposals. That is a separate [ticket](https://github.com/hicommonwealth/commonwealth/issues/3986). We haven't supported it yet.
- Found an intermittent bug that exists in master, so ticketing separate: https://github.com/hicommonwealth/commonwealth/issues/4038

jnaviask reviewed (commented):
Would love to know some constraints around when this hook is safe to use. Will it cause race conditions if e.g. app.chain is undefined? We should also document on the hook itself how to determine when it's complete & update dependent components.

mhagel commented:
> Would love to know some constraints around when this hook is safe to use. Will it cause race conditions if e.g. app.chain is undefined? We should also document on the hook itself how to determine when it's complete & update dependent components.

@jnaviask 
Added a new check for app.preloadingChain, and documented on the hook how it works.

Also added evmos "not supported" message per this discussion:
 https://commonxyz.slack.com/archives/C051XFN57FT/p1684858677872149?thread_ts=1684779785.894519&cid=C051XFN57FT

jnaviask reviewed (changes requested):
Looks like it still triggers a double init despite the hasFetched ref.

mhagel commented:
> Looks like it still triggers a double init despite the hasFetched ref.

I removed the cleanup function. This was resetting the ref when I didn't expect it to. Now just running once.

mhagel commented:
@jnaviask I tested TXs for regen and odin on Frick, and they both behaved as expected.

jnaviask commented:
@mhagel Ian added a new chain init call in #4138 -- can you update this PR to include it, and then we can merge?

mhagel commented on packages/commonwealth/client/scripts/hooks/useInitChainIfNeeded.ts:
```diff
@@ -0,0 +1,26 @@
+
+    chainInit();
+  }, [app?.isAdapterReady]);
+};
```
>@mhagel Ian added a new chain init call in https://github.com/hicommonwealth/commonwealth/pull/4138 -- can you update this PR to include it, and then we can merge?

@jnaviask Yeah this react hook is actually using the same call already. This just makes it a bit more reusable, and makes sure the initData hasn't already been called.

I just retested. Should be good to go.

jnaviask replied:
@mhagel I meant can you add it to the ViewProposalPage, where Ian had to add similar logic in a previous PR? Then we're ready.
